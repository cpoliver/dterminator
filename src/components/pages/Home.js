import React from 'react';
import { Linking, StyleSheet, Text, View } from 'react-native';
import { SocialIcon } from 'react-native-elements';

const socialPlatforms = {
  github: 'https://github.com/cpoliver',
  linkedin: 'https://linkedin.com/in/cpoliver',
  twitter: 'https://twitter.com/cpoliver'
};

const openSocialUrl = (platform) => {
  const url = socialPlatforms[platform];
  Linking.openURL(url);
};

const Home = () => (
  <View style={styles.pageContainer}>
    <View style={styles.titleContainer}>
      <Text style={styles.title}>dterminator</Text>
      <Text style={styles.subtitle}>an app for the indecisive</Text>
    </View>
    <View>
      <Text style={styles.footerText}>made with love by @cpoliver</Text>
      <View style={styles.iconContainer}>
        <SocialIcon type="github" onPress={() => openSocialUrl('github')} light />
        <SocialIcon type="linkedin" onPress={() => openSocialUrl('linkedin')} light />
        <SocialIcon type="twitter" onPress={() => openSocialUrl('twitter')} light />
      </View>
    </View>
  </View>
);

const styles = StyleSheet.create({
  footerText: {
    color: '#d5d4cf',
    fontSize: 18,
    marginBottom: 6
  },
  iconContainer: {
    marginBottom: 12,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  pageContainer: {
    alignItems: 'center',
    backgroundColor: '#201b21',
    flex: 1,
    justifyContent: 'center'
  },
  subtitle: {
    color: '#f6f7fb',
    fontSize: 16
  },
  title: {
    color: '#25c7ee',
    fontSize: 42,
    fontWeight: 'bold'
  },
  titleContainer: {
    alignItems: 'center',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center'
  }
});

export { Home };
