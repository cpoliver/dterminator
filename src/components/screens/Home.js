import React from 'react';
import { Linking, StyleSheet, Text, View } from 'react-native';
import { SocialIcon } from 'react-native-elements';

import { colors } from '../../styles';

const socialPlatforms = {
  github: 'https://github.com/cpoliver',
  linkedin: 'https://linkedin.com/in/cpoliver',
  twitter: 'https://twitter.com/cpoliver'
};

const openSocialUrl = (platform) => {
  const url = socialPlatforms[platform];

  if (url === undefined) {
    throw new Error('invalid param passed to openSocialUrl');
  }

  Linking.openURL(url);
};

const Home = () => (
  <View style={styles.pageContainer}>
    <View style={styles.titleContainer}>
      <Text style={styles.title}>dterminator</Text>
      <Text style={styles.subtitle}>an app for the indecisive</Text>
    </View>
    <View>
      <Text style={styles.footerText}>made with love by @cpoliver</Text>
      <View style={styles.iconContainer}>
        <SocialIcon type="github" onPress={() => openSocialUrl('github')} light />
        <SocialIcon type="linkedin" onPress={() => openSocialUrl('linkedin')} light />
        <SocialIcon type="twitter" onPress={() => openSocialUrl('twitter')} light />
      </View>
    </View>
  </View>
);

const styles = StyleSheet.create({
  footerText: {
    color: colors.light,
    fontSize: 18,
    marginBottom: 6
  },
  iconContainer: {
    marginBottom: 12,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  pageContainer: {
    alignItems: 'center',
    backgroundColor: colors.background,
    flex: 1,
    justifyContent: 'center'
  },
  subtitle: {
    color: colors.lightest,
    fontSize: 16
  },
  title: {
    color: colors.main,
    fontSize: 42,
    fontWeight: 'bold'
  },
  titleContainer: {
    alignItems: 'center',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center'
  }
});

export { Home, openSocialUrl };
