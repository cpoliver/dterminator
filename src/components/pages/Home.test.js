import rn from 'react-native';
import React from 'react';
import { SocialIcon } from 'react-native-elements';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Home, openSocialUrl } from './Home';

const render = () => shallow(
  <Home />
);

describe('the Home component', () => {
  beforeEach(rn.Linking.openURL.mockReset);

  it('should match the snapshot', () => {
    const component = render();

    expect(toJson(component)).toMatchSnapshot();
  });

  describe('pressing the github icon', () => {
    it('should open the github url', () => {
      const icon = render().find(SocialIcon).at(0);
      icon.simulate('press');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://github.com/cpoliver');
    });
  });

  describe('pressing the linkedin icon', () => {
    it('should open the linkedin url', () => {
      const icon = render().find(SocialIcon).at(1);
      icon.simulate('press');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://linkedin.com/in/cpoliver');
    });
  });

  describe('pressing the twitter icon', () => {
    it('should open the twitter url', () => {
      const icon = render().find(SocialIcon).at(2);
      icon.simulate('press');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://twitter.com/cpoliver');
    });
  });
});

describe('the openSocialUrl function', () => {
  beforeEach(rn.Linking.openURL.mockReset);

  describe('when called with "github"', () => {
    it('should open the github url', () => {
      openSocialUrl('github');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://github.com/cpoliver');
    });
  });

  describe('when called with "linkedin"', () => {
    it('should open the linkedin url', () => {
      openSocialUrl('linkedin');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://linkedin.com/in/cpoliver');
    });
  });

  describe('when called with "twitter"', () => {
    it('should open the twitter url', () => {
      openSocialUrl('twitter');

      expect(rn.Linking.openURL).toHaveBeenCalledWith('https://twitter.com/cpoliver');
    });
  });

  describe('when called with an invalid argument', () => {
    it('should throw an error', () => {
      const callWithInvalid = () => openSocialUrl('argument');

      expect(callWithInvalid).toThrow();
      expect(rn.Linking.openURL).not.toHaveBeenCalled();
    });
  });
});
