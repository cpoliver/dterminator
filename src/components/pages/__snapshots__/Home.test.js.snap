// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Home component should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#201b21",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#25c7ee",
          "fontSize": 42,
          "fontWeight": "bold",
        }
      }
    >
      dterminator
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#f6f7fb",
          "fontSize": 16,
        }
      }
    >
      an app for the indecisive
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#d5d4cf",
          "fontSize": 18,
          "marginBottom": 6,
        }
      }
    >
      made with love by @cpoliver
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 12,
        }
      }
    >
      <SocialIcon
        button={false}
        iconColor="white"
        iconSize={24}
        light={true}
        onPress={[Function]}
        raised={true}
        type="github"
      />
      <SocialIcon
        button={false}
        iconColor="white"
        iconSize={24}
        light={true}
        onPress={[Function]}
        raised={true}
        type="linkedin"
      />
      <SocialIcon
        button={false}
        iconColor="white"
        iconSize={24}
        light={true}
        onPress={[Function]}
        raised={true}
        type="twitter"
      />
    </View>
  </View>
</View>
`;
